lib/lazy_object.rb

```diff
@@ -27,4 +27,17 @@ def __target_object__
   def method_missing(method_name, *args, &block)
     __target_object__.send(method_name, *args, &block)
   end
+
+  def present?
+    !blank?
+  end
+
+  def blank?
+    __target_object__.respond_to?(:empty?) ? !!__target_object__.empty? : !__target_object__
+  end
+
+  # Forwards method calls to the target object.
```

Comment:
With active support:

```
[1] pry(main)> obj = LazyObject.new{ nil }
=> #<LazyObject:0x00000008958cf8 @__callable__=#<Proc:0x00000008958c80@(pry):1>>
[2] pry(main)> obj.present?
=> true
```
when expected value is false.

What you should do is to inherit `LazyObject` from `BasicObject`.